drivers/hmis/app/models/hmis/hud/assessment.rb

```diff
@@ -34,10 +34,14 @@ class Hmis::Hud::Assessment < Hmis::Hud::Base
 
   scope :viewable_by, ->(user) do
     enrollment_ids = Hmis::Hud::Enrollment.viewable_by(user).pluck(:id, :EnrollmentID)
-    vieawable_wip = wip_t[:enrollment_id].in(enrollment_ids.map(&:first))
-    viewable_enrollment = as_t[:EnrollmentID].in(enrollment_ids.map(&:second))
+    viewable_wip = wip_t[:enrollment_id].in(enrollment_ids.map(&:first))
+    viewable_completed = as_t[:EnrollmentID].in(enrollment_ids.map(&:second))
 
-    left_outer_joins(:wip).where(vieawable_wip.or(viewable_enrollment))
+    left_outer_joins(:wip).where(viewable_wip.or(viewable_completed))
+  end
+
+  scope :with_role, ->(role) do
+    joins(:assessment_detail).where(assessment_detail: { role: role })
```

Comment:
This is fine, but it might make sense to add a scope to AssessmentDetail
```
scope :with_role ->(role) do
   where(role: role)
end
```
and then merge in the scope in case the role logic changes later.

Reply:
I like that, updated